test/integration/controller/jobs/jobset/jobset_controller_test.go

```diff
@@ -490,6 +491,83 @@ var _ = ginkgo.Describe("JobSet controller", ginkgo.Ordered, ginkgo.ContinueOnFa
 				gomega.Expect(replica2.Labels).ShouldNot(gomega.HaveKey("label1"))
 			})
 		})
+
+		ginkgo.It("Should allow to create jobset with one replicated job replica count 0", func() {
+			mixJobSet := testingjobset.MakeJobSet("mix-jobset", ns.Name).ReplicatedJobs(
+				testingjobset.ReplicatedJobRequirements{
+					Name:        "replicated-job-1",
+					Replicas:    2,
+					Parallelism: 2,
+					Completions: 2,
+				}, testingjobset.ReplicatedJobRequirements{
+					Name:        "replicated-job-2-empty",
+					Replicas:    0,
+					Parallelism: 0,
+					Completions: 0,
+				},
+			).Queue(localQueue.Name).
+				Request("replicated-job-1", corev1.ResourceCPU, "250m").
+				Request("replicated-job-2-empty", corev1.ResourceCPU, "250m").
+				Obj()
+
+			ginkgo.By("create the jobset", func() {
+				gomega.Expect(k8sClient.Create(ctx, mixJobSet)).Should(gomega.Succeed())
+			})
+
+			jobSetLookupKey := types.NamespacedName{Name: mixJobSet.Name, Namespace: ns.Name}
+			ginkgo.By("fetch the jobset and verify it is suspended", func() {
+				gomega.Eventually(func() *bool {
+					createdJobSet1 := &jobsetapi.JobSet{}
+					gomega.Expect(k8sClient.Get(ctx, jobSetLookupKey, createdJobSet1)).Should(gomega.Succeed())
+					return createdJobSet1.Spec.Suspend
+				}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(true)))
+			})
```

Comment:
I don't find this very relevant , if the jobset is not created none of the next steps can succeed. Drop it or move it into the previous `By`